### ISSUE TITLE
Clickhouse Table Constraint Syntax is unsupported

### ISSUE THREAD

**Issue by pelovett** [sqlfluff/sqlfluff#8387]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails to parse table constraints in Clickhouse CREATE TABLE statements.

### Expected Behaviour

Clickhouse allows table definitions to include table-level constraints with two keywords, `ASSUME` and `CHECK`. When parsing tables with these statements, sqlfluff shouldn't error out and they should be parsed into the table object.

https://clickhouse.com/docs/reference/statements/create/table#constraints

### Observed Behaviour

sqlfluff fails to parse clickhouse table constraints correctly. If you try to parse some SQL that uses the valid `ASSUME` keyword, sqlfluff reports a parsing violation:
```
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE
                       | my_table (\n    `date` Date ...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
```

Currently in 
`test/fixtures/dialects/clickhouse/create_table.sql`
[there's a test for this syntax](https://github.com/sqlfluff/sqlfluff/blob/39a9d059d3187689ec40a145bc3fefda7d72937e/test/fixtures/dialects/clickhouse/create_table.sql#L60) with a table constraint :
```SQL
CREATE TABLE table_name
(
    name1 String,
    CONSTRAINT constraint_name_1 CHECK (name1 = 'test')
) ENGINE = engine;
```
 but sqlfluff is parsing it as a column instead of a table:
```
[L:  4, P:  5]      |                column_definition:
[L:  4, P:  5]      |                    naked_identifier:                         'CONSTRAINT'
[L:  4, P: 15]      |                    whitespace:                               ' '
[L:  4, P: 16]      |                    data_type:
[L:  4, P: 16]      |                        data_type_identifier:                 'constraint_name_1'
[L:  4, P: 33]      |                    whitespace:                               ' '
[L:  4, P: 34]      |                    column_constraint_segment:
[L:  4, P: 34]      |                        keyword:                              'CHECK'
[L:  4, P: 39]      |                        whitespace:                           ' '
[L:  4, P: 40]      |                        bracketed:
[L:  4, P: 40]      |                            start_bracket:                    '('
[L:  4, P: 41]      |                            [META] indent:
[L:  4, P: 41]      |                            expression:
[L:  4, P: 41]      |                                column_reference:
[L:  4, P: 41]      |                                    naked_identifier:         'name1'
[L:  4, P: 46]      |                                whitespace:                   ' '
[L:  4, P: 47]      |                                comparison_operator:
[L:  4, P: 47]      |                                    raw_comparison_operator:  '='
[L:  4, P: 48]      |                                whitespace:                   ' '
[L:  4, P: 49]      |                                quoted_literal:               "'test'"
[L:  4, P: 55]      |                            [META] dedent:
[L:  4, P: 55]      |                            end_bracket:                      ')'
```
The `CONSTRAINT` keyword gets improperly labeled as a column identifier and `constraint_name_` is parsed as the data type.

Full output:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_table_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            table_reference:
[L:  1, P: 14]      |                naked_identifier:                             'table_name'
[L:  1, P: 24]      |            newline:                                          '\n'
[L:  2, P:  1]      |            bracketed:
[L:  2, P:  1]      |                start_bracket:                                '('
[L:  2, P:  2]      |                [META] indent:
[L:  2, P:  2]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                column_definition:
[L:  3, P:  5]      |                    naked_identifier:                         'name1'
[L:  3, P: 10]      |                    whitespace:                               ' '
[L:  3, P: 11]      |                    data_type:
[L:  3, P: 11]      |                        data_type_identifier:                 'String'
[L:  3, P: 17]      |                comma:                                        ','
[L:  3, P: 18]      |                newline:                                      '\n'
[L:  4, P:  1]      |                whitespace:                                   '    '
[L:  4, P:  5]      |                column_definition:
[L:  4, P:  5]      |                    naked_identifier:                         'CONSTRAINT'
[L:  4, P: 15]      |                    whitespace:                               ' '
[L:  4, P: 16]      |                    data_type:
[L:  4, P: 16]      |                        data_type_identifier:                 'constraint_name_1'
[L:  4, P: 33]      |                    whitespace:                               ' '
[L:  4, P: 34]      |                    column_constraint_segment:
[L:  4, P: 34]      |                        keyword:                              'CHECK'
[L:  4, P: 39]      |                        whitespace:                           ' '
[L:  4, P: 40]      |                        bracketed:
[L:  4, P: 40]      |                            start_bracket:                    '('
[L:  4, P: 41]      |                            [META] indent:
[L:  4, P: 41]      |                            expression:
[L:  4, P: 41]      |                                column_reference:
[L:  4, P: 41]      |                                    naked_identifier:         'name1'
[L:  4, P: 46]      |                                whitespace:                   ' '
[L:  4, P: 47]      |                                comparison_operator:
[L:  4, P: 47]      |                                    raw_comparison_operator:  '='
[L:  4, P: 48]      |                                whitespace:                   ' '
[L:  4, P: 49]      |                                quoted_literal:               "'test'"
[L:  4, P: 55]      |                            [META] dedent:
[L:  4, P: 55]      |                            end_bracket:                      ')'
[L:  4, P: 56]      |                newline:                                      '\n'
[L:  5, P:  1]      |                [META] dedent:
[L:  5, P:  1]      |                end_bracket:                                  ')'
[L:  5, P:  2]      |            whitespace:                                       ' '
[L:  5, P:  3]      |            engine:
[L:  5, P:  3]      |                keyword:                                      'ENGINE'
[L:  5, P:  9]      |                whitespace:                                   ' '
[L:  5, P: 10]      |                comparison_operator:
[L:  5, P: 10]      |                    raw_comparison_operator:                  '='
[L:  5, P: 11]      |                whitespace:                                   ' '
[L:  5, P: 12]      |                table_engine_function:
[L:  5, P: 12]      |                    function_name:
[L:  5, P: 12]      |                        function_name_identifier:             'engine'
[L:  5, P: 18]      |    statement_terminator:                                     ';'
[L:  5, P: 19]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    [META] end_of_file:
```

### How to reproduce

You can reproduce the parse error by trying to parse a valid ASSUME table constraint:
```bash
echo "CREATE TABLE table_name
(
    name1 String,
    CONSTRAINT constraint_name_1 ASSUME (name1 = 'test')
) ENGINE = engine;" | sqlfluff parse --dialect clickhouse -
```

You can see the improper CHECK constraint parsing with:
```bash
echo "CREATE TABLE table_name
(
    name1 String,
    CONSTRAINT constraint_name_1 CHECK (name1 = 'test')
) ENGINE = engine;" | sqlfluff parse --dialect clickhouse -
```

### Dialect

clickhouse

### Version

sqlfluff, version 4.3.0
Python 3.14.7

### Configuration

NA

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
